<a href="https://colab.research.google.com/github/jeanbaptistehabimanaMIT2026/MIT-projects-2026/blob/main/Jean_Baptiste_HABIMANA_M05688_2026_NumPy_Practice_Work.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

  UNIVERSITY OF LAY ADVENTISTS OF KIGALI
  NAME: Jean Baptiste HABIMANA
  REGNO: M05688/2026

  MODULE: C11712(DATA MINING FOR BUSINESS INTELLIGENCE)
  
  DATE: 28/06/2026

  TASK: NumPy Practice


1. Create the unit_prices array

In [ ]:
import numpy as np

In [ ]:

unit_prices = np.array([450, 1200, 3000, 750, 2100, 980])
print("Unit prices:", unit_prices)


Unit prices: [ 450 1200 3000  750 2100  980]


 2. Print the total (sum)

In [ ]:
print("Total sum:", np.sum(unit_prices))

Total sum: 8480


 3. Print the average (mean), rounded

In [ ]:
print("Average price (rounded):", round(np.mean(unit_prices)))

Average price (rounded): 1413


4. Apply 5% price increase

In [ ]:
increased_prices = unit_prices * 1.05
print("Increased prices:", increased_prices)


Increased prices: [ 472.5 1260.  3150.   787.5 2205.  1029. ]


 5. Boolean mask for prices > 1000

In [ ]:
mask = unit_prices > 1000
print("Prices greater than 1000:", unit_prices[mask])


Prices greater than 1000: [1200 3000 2100]


         OUTPUT:
         unit prices: [450  1200  3000  750  2100  980]
         Total sum: 8480
         Average price (rounded): 1413
         Increased prices: [ 472.5  1260.  3150. 787.5  2205.  1029. ]
         Prices greater than 1000: [1200  3000  2100]


# Comments on NumPy practice on data given for Business Intelligence context for each step:

1. Array creation: loads raw product pricing data into a structure NumPy can analyze at scale

2. Sum :  total revenue potential if one unit of each item is sold ($8,480)

3. Mean:  the average price point (~$1,413) helps benchmark new products or flag outliers

4. 5% increase:  simulates a price adjustment scenario; vectorized operations make this instant across thousands of SKUs

5. Boolean mask : filters the "premium" segment (>$1,000), useful for targeted marketing or tiered discount strategies

              Thanks.
